Add support for django 3.2

pgammans opened the pull request:
This commit add support for django 3.2, 

The current code is broken by django commit 9d519d3dc4e5bd1d9ff3806b44624c3e487d61c1 which fixes the django issues #31755

millerdev commented on django_cte/meta.py:
```diff
@@ -47,12 +48,36 @@ def _ref(self):
         return self._ref.target
 
+    @property
+    def _output_field_or_none(self):
```
Where is this used?

pgammans replied:
This is called from  django.db.models.expressions.BaseExpression.get_source_fields
and is a replacement function for django.db.models.expressions.BaseExpression._output_field_or_none
but ensures that we dont try to fetch the field type of a column without a query.

millerdev reviewed (commented):
Thanks for the contribution, @pgammans. Did django-cte tests fail when run against Django 3.2? If not, please add a test that would have failed prior to this fix.

pgammans commented:
Yes the following four test failed:
- test_circular_ref_error (tests.test_recursive.TestRecursiveCTE)
- test_pickle_recursive_cte_queryset (tests.test_recursive.TestRecursiveCTE)
- test_recursive_cte_query (tests.test_recursive.TestRecursiveCTE)
- test_recursive_cte_reference_in_condition (tests.test_recursive.TestRecursiveCTE)

with the error:  ValueError: cannot resolve 'cte.depth' in recursive CTE setup. Hint: use ExpressionWrapper(cte.col.depth, output_field=...)

I havent added the test for 3.2 to the travis.yml file do you want me to do this?

Also i have removed the code
 - Function CTEColumn.output_field_or_none:  As I cannot remember why it was needed and have not test that uses that path
 - Function CTEColumn.output_field setter:  Again i cannot find why I needed to add this

I sure i will find why i added them later but I will open another PR with test when i find the issues.

millerdev reviewed (approved):
Thanks! We'll merge after tests pass.

pgammans commented:
Thanks, I missed a exclude in the test matrix for python 2.7 and Django 3.2
